core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -790,34 +823,33 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     }
 
     while (cur ne null) {
-      if (RuntimeFlags.opSupervision(runtimeFlags)) {
-        self.getSupervisor().onEffect(self, cur)
-      }
+
+      cur = drainQueueWhileRunning(runtimeFlags, lastTrace, cur)
 
       val nextTrace = cur.trace
       if (nextTrace ne Trace.empty) lastTrace = nextTrace
 
-      cur = drainQueueWhileRunning(runtimeFlags, lastTrace, cur)
+      if (RuntimeFlags.opSupervision(runtimeFlags)) {
```

Comment:
@mschuwalow FYI, I moved this to after drainQueueWhileRunning because the runtime wasn't necessarily about to execute the effect reported to the supervisor. This may impact your profiler (I suspect not in any serious way though).